### PR TITLE
[SPARK-14301][Examples] Java examples code merge and clean up.

### PR DESCRIPTION
**Pull request by yongtang**
## What changes were proposed in this pull request?

This fix tries to remove duplicate Java code in examples/mllib and examples/ml. The following changes have been made:

```
deleted: ml/JavaCrossValidatorExample.java (duplicate of JavaModelSelectionViaCrossValidationExample.java)
deleted: ml/JavaTrainValidationSplitExample.java (duplicated of JavaModelSelectionViaTrainValidationSplitExample.java)
deleted: mllib/JavaFPGrowthExample.java (duplicate of JavaSimpleFPGrowth.java)
deleted: mllib/JavaLDAExample.java (duplicate of JavaLatentDirichletAllocationExample.java)
deleted: mllib/JavaKMeans.java (merged with JavaKMeansExample.java)
deleted: mllib/JavaLR.java (duplicate of JavaLinearRegressionWithSGDExample.java)
updated: mllib/JavaKMeansExample.java (merged with mllib/JavaKMeans.java)
```
## How was this patch tested?

Existing tests passed.

### CODE REVIEW

**Comment by holdenk** [comment]
Yay deleting code! :) Verified none of the removed examples are currently directly referenced in includes under docs/ - @yongtang still might be good for test purposes to build the docs and verify no new warnings just to be on the safe side.

**Comment by srowen** [comment]
Jenkins test this please

**Comment by yongtang** [comment]
Thanks @holdenk. Just tried with `SKIP_API=1 jekyll build` and the docs builds fine.

**Comment by holdenk** [comment]
Awesome :) A few minor things found while reviewing:

One thing I noticed while looking at the differences between the code we are deleting and the code we are keeping is that some of the code we are deleting had a some useful javadoc on how to run the example and where it was from (e.g. `JavaTrainValidationSplitExample.java`) we might want to copy over.

Also I think the reference for what `JavaSimpleTextClassificationPipeline.java` duplicated is pointing to the wrong file since it doesn't do the cross validator stuff (although it looks like the cross validator is a super set so maybe its ok to kill anyways - but if we do kill it we should probably rename the new  to include something like "WithTextClassificationPipeline" one so people looking for a sample text pipeline 
can still find it)

**Comment by jkbradley** [comment]
Please do not delete JavaDeveloperApiExample.java   AFAIK, this is the only code example we have for Java developers who want to build custom ML pipelines.

**Comment by yongtang** [comment]
Thanks @jkbradley @holdenk.

Both `JavaDeveloperApiExample.java` and `JavaSimpleTextClassificationPipeline.java` have been restored.

Some of the useful comments have also been consolidated into the merged files.

**Comment by srowen** [comment]
Jenkins retest this please.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaModelSelectionViaTrainValidationSplitExample.java [hunk 1]
```diff
@@ -33,6 +33,16 @@
 
 /**
  * Java example for Model Selection via Train Validation Split.
+ *
+ * A simple example demonstrating model selection using TrainValidationSplit.
```
Comment (MLnick) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaModelSelectionViaTrainValidationSplitExample.java]: We can consolidate this comment line with the one above, this one is a bit redundant. Maybe something like `Java example demonstrating model selection using TrainValidationSplit"

Comment (yongtang) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaModelSelectionViaTrainValidationSplitExample.java]: Thanks @MLnick the comment lines have been updated.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**